File: php/src/Advertisements/Advertisement/Domain/ValueObjects/ActiveAdvertisements.php
<?php
declare(strict_types=1);

namespace Demo\App\Advertisements\Advertisement\Domain\ValueObjects;

readonly class ActiveAdvertisements
{
    private function __construct(private int $activeAdvertisements)
    {
    }

    public static function fromInt(int $activeAdvertisements): self
    {
        return new self($activeAdvertisements);
    }

    public function value(): int
    {
        return $this->activeAdvertisements;
    }
}

File: kotlin/src/main/kotlin/advertisements/advertisement/domain/value_object/ActiveAdvertisements.kt
package advertisements.advertisement.domain.value_object

class ActiveAdvertisements private constructor(private val activeAdvertisements: Int) {

    companion object {
        fun fromInt(activeAdvertisements: Int): ActiveAdvertisements {
            return ActiveAdvertisements(activeAdvertisements)
        }
    }

    fun value(): Int {
        return activeAdvertisements
    }
}